module/documents/activity/attack.mjs

```diff
@@ -179,16 +179,15 @@ export default class AttackActivity extends ActivityMixin(AttackActivityData) {
       _applyDeprecatedD20Configs(rollConfig, dialogConfig, messageConfig, oldConfig);
     }
 
-    const createMessage = messageConfig.create !== false;
-    messageConfig.create = false;
     const rolls = await CONFIG.Dice.D20Roll.build(rollConfig, dialogConfig, messageConfig);
     if ( !rolls.length ) return null;
-    if ( createMessage ) {
+    if ( messageConfig.document ) {
```

Comment:
We could use the same technique as `rollRecharge` and perform `buildConfigure` and `buildEvaluate` separately, then add the flags before calling `buildPost`. That saves a round-trip to the server and a re-render.